tests/sentry/hybrid_cloud/test_organization.py

```diff
@@ -280,6 +288,100 @@ def test_options():
+    assert replicated.provider == auth_provider.provider
+    assert replicated.config == auth_provider.config
+    assert replicated.default_role == auth_provider.default_role
+    assert replicated.default_global_access == auth_provider.default_global_access
+    assert replicated.scim_enabled == auth_provider.flags.scim_enabled
+    assert replicated.allow_unlinked == auth_provider.flags.allow_unlinked
+
+    with assume_test_silo_mode(SiloMode.CONTROL):
+        auth_provider.provider = "new_provider"
+        auth_provider.flags.scim_enabled = not auth_provider.flags.scim_enabled
+        auth_provider.save()
+
+    with assume_test_silo_mode(SiloMode.REGION):
+        replicated = RegionReplicatedAuthProvider.objects.get(organization_id=org.id)
+
+    assert replicated.auth_provider_id == auth_provider.id
+    assert replicated.provider == auth_provider.provider
+    assert replicated.scim_enabled == auth_provider.flags.scim_enabled
+
+    serialized = serialize_auth_provider(auth_provider)
+    serialized.organization_id = 99999
+
+    # Should still succeed despite non existent organization
+    organization_service.upsert_replicated_auth_provider(auth_provider=serialized, region_name="us")
+
+
+@django_db_all(transaction=True)
+@all_silo_test(stable=True)
+def test_replicate_auth_identity():
+    user = Factories.create_user()
+    org = Factories.create_organization(owner=user)
+
+    with assume_test_silo_mode(SiloMode.REGION):
+        assert RegionReplicatedAuthIdentity.objects.count() == 0
+
+    with assume_test_silo_mode(SiloMode.CONTROL):
+        auth_provider = AuthProvider.objects.create(
+            organization_id=org.id, provider="abc", config={"a": 1}
+        )
+        auth_identity = AuthIdentity.objects.create(
+            user=user, auth_provider=auth_provider, ident="some-ident", data={"b": 2}
+        )
+
+    with assume_test_silo_mode(SiloMode.REGION):
+        replicated = RegionReplicatedAuthIdentity.objects.get(
+            ident=auth_identity.ident, auth_provider_id=auth_provider.id
+        )
+
+    assert replicated.auth_identity_id == auth_identity.id
+    assert replicated.auth_provider_id == auth_identity.auth_provider_id
+    assert replicated.user_id == auth_identity.user_id
+    assert replicated.data == auth_identity.data
+    assert replicated.ident == auth_identity.ident
+
+    with assume_test_silo_mode(SiloMode.CONTROL):
+        auth_identity.data = {"v": "new data"}
+        auth_identity.save()
+
+    with assume_test_silo_mode(SiloMode.REGION):
+        replicated = RegionReplicatedAuthIdentity.objects.get(
+            ident=auth_identity.ident, auth_provider_id=auth_provider.id
+        )
+
+    assert replicated.auth_identity_id == auth_identity.id
+    assert replicated.data == auth_identity.data
+
+    serialized = serialize_auth_identity(auth_identity)
+    serialized.id = 99999
+
+    # Should still succeed despite non existent organization
```

Comment:
need to update this comment -- actually in this particular case we want to fail, so as to trigger retry, to 'reorder things' until the unique columns can be satisfied.